lib/routes/routeBackbeat.js

```diff
@@ -304,12 +304,14 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
             // use original data locations
             omVal.location = objMd.location;
         }
-        // specify both 'versioning' and 'versionId' to create a "new"
-        // version (updating master as well) but with specified
-        // versionId
+        const { isNFS, status } =  omVal.replicationInfo;
+        // Specify both 'versioning' and 'versionId' to create a "new" version
+        // (updating master as well) but with specified versionId. If the object
+        // is from a source bucket without versioning (i.e. NFS), then we want
+        // to create a version for the replica object even though none was
+        // provided in the object metadata value.
         const options = {
-            versioning: omVal.versionId !== undefined ||
-                omVal.replicationInfo.isNFS === true,
+            versioning: isNFS !== true || status === 'REPLICA',
```

Comment:
Do you want to reset ```omVal.replicationInfo.isNFS``` when ```status === 'REPLICA'```? You probably don't want to propagate this property to the destination object.

Reply:
I think it should be okay as-is because we are not altering any of the other `replicationInfo` properties sent along from the source. That way the metadata of the objects remain identical.

Reply:
I can think of a case though where the destination bucket also has CRR configured on it and is thus a source for another bucket in which case maybe we do want to update this value.

Reply:
I was thinking about that particular use case

Reply:
Okay, just to track this: we do not replicate 'REPLICA' objects (as per AWS docs https://docs.aws.amazon.com/AmazonS3/latest/dev/crr-what-is-isnot-replicated.html), but will this leave as-is.